src/Microsoft.Framework.Runtime.Hosting/Json/JsonDeserializerResource.cs

```diff
@@ -7,12 +7,12 @@ internal class JsonDeserializerResource
     {
         internal static string Format_IllegalCharacter(int value)
         {
-            return string.Format("Illegal character {0} {0:X4}.", (char)value, value);
+            return string.Format("Illegal character '{0}' [Unicode Hexadecimal {1:X4}].", (char)value, value);
         }
 
         internal static string Format_IllegalTrailingCharacterAfterLiteral(int value, string literal)
         {
-            return string.Format("Illegal character {0} ({1:X4}) after the literal name {2}.",
+            return string.Format("Illegal character '{0}' [Unicode Hexadecimal {1:X4}] after the literal name {2}.",
```

Comment:
_string interpolation_ doesn't apply here, because the _value_ need to be converted to _char_